app-frontend/src/app/pages/imports/datasources/list/list.controller.js

```diff
@@ -1,45 +1,33 @@
 /* global BUILDCONFIG */
 
 class DatasourceListController {
-    constructor($state, datasourceService, modalService, $filter) {
+    constructor(
+        $scope, $state, $stateParams, $filter,
+        datasourceService, modalService, paginationService
+    ) {
         'ngInject';
-        this.$state = $state;
-        this.$filter = $filter;
-        this.datasourceService = datasourceService;
-        this.modalService = modalService;
+        $scope.autoInject(this, arguments);
+        this.pageSize = 10;
     }
 
     $onInit() {
         this.BUILDCONFIG = BUILDCONFIG;
-        this.datasources = {};
-        this.currentPage = 1;
-        this.pageSize = 10;
-        this.loadDatasources();
-    }
-
-    $onDestroy() {
-
+        this.fetchPage();
     }
 
     shouldShowPlaceholder() {
-        return !this.isLoadingDatasources &&
-            this.datasources.count &&
-            this.datasources.count === 0;
+        return !this.currentQuery &&
+            !this.fetchError &&
+            (!this.search || !this.search.length) &&
+            this.datasources &&
+            this.datasources.length === 0;
     }
 
-    shouldShowList() {
-        return !this.isLoadingDatasources &&
-            this.datasources.count &&
-            this.datasources.count > 0 &&
-            !this.searchString;
-    }
-
-    shouldShowPagination() {
-        return !this.isLoadingDatasources &&
-            !this.isErrorLoadingDatasources &&
-            this.datasources.count &&
-            this.datasources.count > this.pageSize &&
-            !this.searchString;
+    shouldShowEmptySearch() {
+        return !this.currentQuery &&
+            !this.fetchError &&
+            this.search && this.search.length &&
+            this.datasources && !this.datasources.length;
     }
 
     updateQueryParameters() {
```

Comment:
It seems like similar method has been moved into the pagination service. I would remove this function here.

Reply:
nope, pagination service returns info about the response, but doesn't store info about current query progress or search (I show a different element when returning an empty search)

Reply:
https://github.com/raster-foundry/raster-foundry/blob/develop/app-frontend/src/app/services/common/pagination.service.js#L36-L46

Reply:
yea I know that. this function handles query / search state, not location parameters

Reply:
:man_facepalming: I get what you guys are saying now.  was looking at the wrong function. sorry.

Reply:
🎉